test/test_varlen_attention.py

```diff
@@ -81,59 +120,116 @@ def create_variable_length_batch(shape: VarlenShape, device: torch.device, dtype
 
     return {
-        'seq_lengths': seq_lengths,
-        'cu_seq': cu_seq,
-        'x_packed': x_packed,
-        'x_padded': x_padded,
-        'max_len': max_len,
-        'total_tokens': total_tokens
+        "seq_lengths": seq_lengths,
+        "cu_seq": cu_seq,
+        "x_packed": x_packed,
+        "x_padded": x_padded,
+        "max_len": max_len,
+        "total_tokens": total_tokens,
     }
 
 
 class TestVarlenAttention(NNTestCase):
-
-    @unittest.skipIf(not PLATFORM_SUPPORTS_FLASH_ATTENTION, "Flash Attention not supported")
+    @unittest.skipIf(
+        not PLATFORM_SUPPORTS_FLASH_ATTENTION, "Flash Attention not supported"
+    )
     @parametrize("dtype", [torch.bfloat16, torch.float16])
     def test_basic_functionality(self, device, dtype):
         torch.manual_seed(42)
 
         shape = VarlenShape(batch_size=2, max_seq_len=512, embed_dim=1024, num_heads=16)
 
-        attention_block = AttentionBlock(shape.embed_dim, shape.num_heads, device, dtype)
+        attention_block = AttentionBlock(
+            shape.embed_dim, shape.num_heads, device, dtype
+        )
 
         total_tokens = shape.batch_size * shape.max_seq_len
-        x_packed = torch.randn(total_tokens, shape.embed_dim, device=device, dtype=dtype)
-        cu_seq = torch.tensor([0, shape.max_seq_len, total_tokens], device=device, dtype=torch.int32)
+        x_packed = torch.randn(
+            total_tokens, shape.embed_dim, device=device, dtype=dtype
+        )
+        cu_seq = torch.tensor(
+            [0, shape.max_seq_len, total_tokens], device=device, dtype=torch.int32
+        )
 
-        output = attention_block.forward_varlen(x_packed, cu_seq, shape.max_seq_len, is_causal=False)
+        output = attention_block.forward_varlen(
+            x_packed, cu_seq, shape.max_seq_len, is_causal=False
+        )
 
         self.assertEqual(output.shape, (total_tokens, shape.embed_dim))
         self.assertEqual(output.device, torch.device(device))
         self.assertEqual(output.dtype, dtype)
 
-    @unittest.skipIf(not PLATFORM_SUPPORTS_FLASH_ATTENTION, "Flash Attention not supported")
+    @unittest.skipIf(
+        not PLATFORM_SUPPORTS_FLASH_ATTENTION, "Flash Attention not supported"
+    )
+    @parametrize("dtype", [torch.bfloat16, torch.float16])
+    def test_custom_op_registration(self, device, dtype):
+        torch.manual_seed(42)
+
+        shape = VarlenShape(batch_size=2, max_seq_len=512, embed_dim=1024, num_heads=16)
+
+        attention_block = AttentionBlock(
+            shape.embed_dim, shape.num_heads, device, dtype
+        )
+
+        total_tokens = shape.batch_size * shape.max_seq_len
+        x_packed = torch.randn(
+            total_tokens, shape.embed_dim, device=device, dtype=dtype
+        )
+        cu_seq = torch.tensor(
+            [0, shape.max_seq_len, total_tokens], device=device, dtype=torch.int32
+        )
+
+        compiled_forward = torch.compile(
+            attention_block.forward_varlen, backend="eager"
```

Comment:
I'd specify `fullgraph=True` here to ensure we're not supporting this by graph breaking